lib/rubocop/cop/rails/date.rb

```diff
@@ -109,6 +109,16 @@ def safe_chain?(chain)
           (chain & bad_methods).empty? || !(chain & good_methods).empty?
         end
 
+        def safe_to_time?(node)
+          receiver, _method_name, *args = *node
+          if receiver.str_type?
+            zone_regexp = /[\+-]{1}[\d:]+$/
```

Comment:
Also you can probably do `\z` instead of `$`. `$` means end of line. `\z` means end of string.

```ruby
irb(main):001:0> "John\nJacob\nJingleheimer\nSchmidt" =~ /^Jacob$/
=> 5
irb(main):002:0> "John\nJacob\nJingleheimer\nSchmidt" =~ /^Jacob\z/
=> nil